specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -169,78 +237,103 @@ function Fields({
   readonly fields: GetSet<Formatter['definition']['fields'][number]['fields']>;
   readonly onDelete: (() => void) | undefined;
 }): JSX.Element {
+  const [displayFormatter, setDisplayFormatter] = React.useState(false);
   return (
-    <table
-      /*
-       * REFACTOR: replace min-w-[35rem] with a container query that replaces
-       *   table layout with list layout
-       */
-      className={`
-        grid-table min-w-[35rem] grid-cols-[min-content_auto_auto_min-content]
-        gap-2
-      `}
-    >
-      <thead>
-        <tr>
-          <th>{resourcesText.separator()}</th>
-          <th>{schemaText.field()}</th>
-          <th>{resourcesText.formatter()}</th>
-          <td />
-        </tr>
-      </thead>
-      <tbody>
-        {fields.map((field, index) => (
-          <Field
-            field={[
-              field,
-              (field): void => setFields(replaceItem(fields, index, field)),
-            ]}
-            key={index}
-            table={table}
-            onRemove={(): void => setFields(removeItem(fields, index))}
-          />
-        ))}
-        <tr>
-          <td className="col-span-3 !gap-2">
-            <Button.Success
-              onClick={(): void =>
-                setFields([
-                  ...fields,
-                  {
-                    separator: localized(' '),
-                    aggregator: undefined,
-                    formatter: undefined,
-                    fieldFormatter: undefined,
-                    field: undefined,
-                  },
-                ])
-              }
-            >
-              {resourcesText.addField()}
-            </Button.Success>
-            {typeof handleDelete === 'function' && (
-              <Button.Danger onClick={handleDelete}>
-                {resourcesText.deleteDefinition()}
-              </Button.Danger>
-            )}
-          </td>
-          <td />
-        </tr>
-      </tbody>
-    </table>
+    <>
+      {fields.length === 0 ? null : (
```

Comment:
it's considered bad practice/code smell when the component decides for itself if it should be rendered or not - such power should lie in it's parent (i.e if the parent decided than the child should be displayed, than the child should be displayed - otherwise, if both parent decides that, and child decides that, there could be a bug where parent wants to display the child, but child decided not to show up).
could you please move this check into the parent? (i.e `{fields.length > 0 && <Fields ... />`)

of course there are exceptions to all rules - the exception to this one is when the child component is called in multiple places with the same condition, or if condition is not available outside of the child (i.e child is fetching some data, and while loading doesn't render anything to the screen) - neither of those apply here

Reply:
the field.length is just for the table not for the rest (Add button etc) 
If I move it to the parent than we are missing those